utils/weighted_random.go

```diff
@@ -0,0 +1,155 @@
+package utils
```

Comment:
Since you have types, please move to a separate package...

Reply:
done